contrib/build-release.sh

```diff
@@ -0,0 +1,49 @@
+#! /bin/sh
+set -e
+
+if [ "$(git status --porcelain -u no)" != "" ]; then
+    echo "Not a clean git directory" >&2
+    exit 1
+fi
+
+VERSION=$(git tag --points-at HEAD)
+if [ "$VERSION" = "" ]; then
+    echo "No tagged version at HEAD?" >&2
+    exit 1
+fi
+
+# Make sure repo is clean.
+make distclean
+
+for platform in Fedora-28-amd64 Ubuntu-16.04-amd64 Ubuntu-16.04-i386; do
+    case $platform in
+	Fedora-28-amd64)
+	    DOCKERFILE=contrib/Dockerfile.builder.fedora
+	    TAG=fedora
+	    ;;
+	Ubuntu-16.04-amd64)
+	    DOCKERFILE=contrib/Dockerfile.builder
+	    TAG=ubuntu-amd64
+	    ;;
+	Ubuntu-16.04-i386)
+	    DOCKERFILE=contrib/Dockerfile.builder.i386
+	    TAG=ubuntu-i386
+	    ;;
+	*)
+	    echo "No Dockerfile for $platform" >&2
+	    exit 1
+    esac
+
+    docker build -f $DOCKERFILE -t $TAG .
+    docker run --rm=true -v "$(pwd)":/build -w /build $TAG ./configure
```

Comment:
We could save on the make distclean (and make really sure we don't have stale things around by `git clone`ing in the docker container, merge the steps below and just delete the container afterwards. That way we don't rely on `distclean` getting absolutely everything.